### PR TITLE
feat: Issue connection changed to unidirectional relation

### PR DESCRIPTION
**Pull request by shuofan**
#### What this PR does / why we need it:
Issue connection changed to unidirectional relation

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMTMxLDExMTUsMTE0Nl0sImFzc2lnbmVlIjpbIjEwMDEwNzMiXX0%3D&id=291693&iterationID=1115&pId=0&type=BUG)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  Issue connection changed to unidirectional relation            |
| 🇨🇳 中文    |     事项关联改为单向关系         |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/dop/endpoints/issue_related.go [hunk 1]
```diff
@@ -101,33 +101,38 @@ func (e *Endpoints) GetIssueRelations(ctx context.Context, r *http.Request, vars
...
-		}
-		for _, issue := range relatedIssues {
-			userIDs = append(userIDs, issue.Creator, issue.Assignee)
-		}
-		userIDs = strutil.DedupSlice(userIDs, true)
-
 		if i == apistructs.IssueRelationInclusion {
+			relatingIssues, err := e.issue.GetIssuesByIssueIDs(relatingIssueIDs, identityInfo)
+			if err != nil {
+				return apierrors.ErrGetIssueRelations.InternalError(err).ToResp(), nil
```
Comment (Effet) [inline modules/dop/endpoints/issue_related.go:107]: I suggest to encapsulate a func

**Review by Effet — CHANGES_REQUESTED** [review]

File: modules/dop/dao/issue_issue_relation.go [hunk 2]
```diff
@@ -49,8 +49,10 @@ func (client *DBClient) IssueRelationsExist(issueRelation *IssueRelation, relate
 		}
 		return count > 0, nil
 	}
+	sql := client.Table("dice_issue_relation").Where("type = ?", issueRelation.Type)
 	var count int64
-	if err := client.Table("dice_issue_relation").Where("issue_id = ? and related_issue in (?) and type = ?", issueRelation.IssueID, relatedIssues, issueRelation.Type).Count(&count).Error; err != nil {
+	if err := sql.Where("issue_id = ? and related_issue in (?) or issue_id in (?) and related_issue = ?", issueRelation.IssueID, relatedIssues, relatedIssues, issueRelation.IssueID).
```
[the remark this answers is not in the document]
Comment (shuofan) [reply modules/dop/dao/issue_issue_relation.go:54]: sure

File: modules/dop/endpoints/issue_related.go (hunk 1 above)
Comment (shuofan) [reply modules/dop/endpoints/issue_related.go]: done

File: modules/dop/dao/issue_issue_relation.go [hunk 3]
```diff
@@ -104,7 +106,8 @@ func (client *DBClient) GetRelatedIssues(issueID uint64, relationType []string)
 
 // DeleteIssueRelation 删除两条issue之间的关联关系
 func (client *DBClient) DeleteIssueRelation(issueID, relatedIssueID uint64, relationTypes []string) error {
-	query := client.Table("dice_issue_relation").Where("issue_id = ?", issueID).Where("related_issue = ?", relatedIssueID)
+	query := client.Table("dice_issue_relation").Where("issue_id = ? and related_issue = ? or issue_id = ? and related_issue = ?",
+		issueID, relatedIssueID, relatedIssueID, issueID)
```
Comment (Effet) [inline modules/dop/dao/issue_issue_relation.go:110]: also add some (, )

**Review by Effet — CHANGES_REQUESTED** [review]

File: modules/dop/dao/issue_issue_relation.go (hunk 3 above)
Comment (shuofan) [reply modules/dop/dao/issue_issue_relation.go]: done

**Review by Effet — APPROVED** [review]

**Pull request merged**